plugins/obs-ffmpeg/jim-nvenc.c

```diff
@@ -997,8 +1005,9 @@ static bool init_encoder(struct nvenc_data *enc, bool hevc,
 
 		blog(LOG_WARNING, "[jim-nvenc] init_specific_encoder failed, "
 				  "trying again without Psycho Visual Tuning");
+		reinit_session(enc);
```

Comment:
Can you do this instead?
```
blog(LOG_WARNING, "[jim-nvenc] init_specific_encoder failed, "
		  "trying again without Psycho Visual Tuning");
nv.nvEncDestroyEncoder(enc->session);
enc->session = NULL;
if (!init_session(enc) ||
    !init_specific_encoder(enc, hevc, settings, encoder, bf,
			   false)) {
	return false;
}
```

Reply:
yea, it seem like that I have ignored the init_session return value here.
thx, I will make this changes soon.

Reply:
I'm also not a fan of stubby little helper functions if they're not called more than once. I'd rather see the logic directly in context.

Thanks for the bugfix.

Reply:
Yes, I agree with you.
Maybe I think using `reinit` help function can help understanding and make the code more readable though only used once.